--- services/haveged.te	1970-01-01 01:00:00.000000000 +0100
+++ services/haveged.te	2011-07-17 19:28:40.508000843 +0200
@@ -0,0 +1,19 @@
+policy_module(haveged, 1.0.0)
+
+########################################
+#
+# Declarations
+#
+
+type haveged_t;
+type haveged_exec_t;
+init_daemon_domain(haveged_t, haveged_exec_t)
+
+type haveged_var_run_t;
+files_pid_file(haveged_var_run_t)
+
+########################################
+#
+# haveged local policy
+#
+kernel_read_kernel_sysctls(haveged_t)
--- services/haveged.fc	1970-01-01 01:00:00.000000000 +0100
+++ services/haveged.fc	2011-07-17 17:55:56.431000683 +0200
@@ -0,0 +1 @@
+/usr/sbin/haveged			--	gen_context(system_u:object_r:haveged_exec_t,s0)
