--- services/postfix.fc	2010-08-03 15:11:07.000000000 +0200
+++ services/postfix.fc	2011-03-03 14:20:12.528999997 +0100
@@ -32,6 +32,9 @@
 /usr/lib/postfix/virtual --	gen_context(system_u:object_r:postfix_virtual_exec_t,s0)
 ')
 /etc/postfix/postfix-script.* -- gen_context(system_u:object_r:postfix_exec_t,s0)
+ifdef(`distro_gentoo',`
+/usr/lib(64)?/postfix/[^\.]*	--	gen_context(system_u:object_r:bin_t,s0)
+')
 /etc/postfix/prng_exch	--	gen_context(system_u:object_r:postfix_prng_t,s0)
 /usr/sbin/postalias	--	gen_context(system_u:object_r:postfix_master_exec_t,s0)
 /usr/sbin/postcat	--	gen_context(system_u:object_r:postfix_master_exec_t,s0)
