--- refpolicy.orig/policy/modules/services/networkmanager.te	2010-12-20 20:39:58.518000002 +0100
+++ refpolicy/policy/modules/services/networkmanager.te	2010-12-28 16:22:46.593000059 +0100
@@ -28,6 +28,9 @@
 type wpa_cli_exec_t;
 init_system_domain(wpa_cli_t, wpa_cli_exec_t)
 
+type wpa_cli_tmp_t;
+files_tmp_file(wpa_cli_tmp_t)
+
 ########################################
 #
 # Local policy
@@ -59,6 +62,10 @@
 manage_sock_files_pattern(NetworkManager_t, NetworkManager_tmp_t, NetworkManager_tmp_t)
 files_tmp_filetrans(NetworkManager_t, NetworkManager_tmp_t, { sock_file file })
 
+manage_files_pattern(wpa_cli_t, wpa_cli_tmp_t, wpa_cli_tmp_t)
+manage_sock_files_pattern(wpa_cli_t, wpa_cli_tmp_t, wpa_cli_tmp_t)
+files_tmp_filetrans(wpa_cli_t, wpa_cli_tmp_t, { sock_file file })
+
 manage_dirs_pattern(NetworkManager_t, NetworkManager_var_lib_t, NetworkManager_var_lib_t)
 manage_files_pattern(NetworkManager_t, NetworkManager_var_lib_t, NetworkManager_var_lib_t)
 files_var_lib_filetrans(NetworkManager_t, NetworkManager_var_lib_t, dir)
@@ -125,6 +132,7 @@
 init_read_utmp(NetworkManager_t)
 init_dontaudit_write_utmp(NetworkManager_t)
 init_domtrans_script(NetworkManager_t)
+init_domtrans_script(wpa_cli_t)
 
 auth_use_nsswitch(NetworkManager_t)
 
@@ -265,6 +273,10 @@
 	vpn_signull(NetworkManager_t)
 ')
 
+ifdef(`distro_gentoo',`
+	sysnet_domtrans_dhcpc(wpa_cli_t)
+')
+
 ########################################
 #
 # wpa_cli local policy
@@ -276,7 +288,7 @@
 allow wpa_cli_t NetworkManager_t:unix_dgram_socket sendto;
 
 manage_sock_files_pattern(wpa_cli_t, NetworkManager_tmp_t, NetworkManager_tmp_t)
-files_tmp_filetrans(wpa_cli_t, NetworkManager_tmp_t, sock_file)
+#files_tmp_filetrans(wpa_cli_t, NetworkManager_tmp_t, sock_file)
 
 list_dirs_pattern(wpa_cli_t, NetworkManager_var_run_t, NetworkManager_var_run_t)
 rw_sock_files_pattern(wpa_cli_t, NetworkManager_var_run_t, NetworkManager_var_run_t)
--- refpolicy.orig/policy/modules/services/networkmanager.if	2010-12-20 20:39:58.529000002 +0100
+++ refpolicy/policy/modules/services/networkmanager.if	2010-12-20 22:50:34.558000007 +0100
@@ -191,3 +191,48 @@
 	files_search_pids($1)
 	allow $1 NetworkManager_var_run_t:file read_file_perms;
 ')
+
+########################################
+## <summary>
+##      Execute wpa_cli in the wpa_cli domain.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed to transition.
+##      </summary>
+## </param>
+#
+interface(`networkmanager_domtrans_wpa_cli',`
+        gen_require(`
+                type wpa_cli_t, wpa_cli_exec_t;
+        ')
+
+        corecmd_search_bin($1)
+        domtrans_pattern($1, wpa_cli_exec_t, wpa_cli_t)
+')
+
+########################################
+## <summary>
+##      Execute wpa cli in the wpa_cli domain, and
+##      allow the specified role the wpa_cli domain.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed to transition.
+##      </summary>
+## </param>
+## <param name="role">
+##      <summary>
+##      Role allowed access.
+##      </summary>
+## </param>
+## <rolecap/>
+#
+interface(`networkmanager_run_wpa_cli',`
+        gen_require(`
+                type wpa_cli_exec_t;
+        ')
+
+        networkmanager_domtrans_wpa_cli($1)
+        role $2 types wpa_cli_t;
+')
--- refpolicy.orig/policy/modules/services/networkmanager.fc	2010-12-20 20:39:58.494000002 +0100
+++ refpolicy/policy/modules/services/networkmanager.fc	2010-12-24 14:33:06.956000260 +0100
@@ -24,3 +24,4 @@
 /var/run/nm-dhclient.*			gen_context(system_u:object_r:NetworkManager_var_run_t,s0)
 /var/run/wpa_supplicant(/.*)?		gen_context(system_u:object_r:NetworkManager_var_run_t,s0)
 /var/run/wpa_supplicant-global	-s	gen_context(system_u:object_r:NetworkManager_var_run_t,s0)
+/tmp/wpa_ctrl_(.*)?		--	gen_context(system_u:object_r:wpa_cli_tmp_t,s0)
