--- services/zabbix.te	2010-12-13 15:11:02.000000000 +0100
+++ services/zabbix.te	2011-06-12 09:38:04.377000217 +0200
@@ -28,6 +28,7 @@
 allow zabbix_t self:capability { setuid setgid };
 allow zabbix_t self:fifo_file rw_file_perms;
 allow zabbix_t self:unix_stream_socket create_stream_socket_perms;
+allow zabbix_t self:sem { create unix_write unix_read read write associate destroy }; #mutex requirement for log file
 
 # log files
 allow zabbix_t zabbix_log_t:dir setattr;
--- services/zabbix.fc	2010-08-03 15:11:09.000000000 +0200
+++ services/zabbix.fc	2011-06-12 09:27:16.895000179 +0200
@@ -1,6 +1,7 @@
 /etc/rc\.d/init\.d/zabbix --	gen_context(system_u:object_r:zabbix_initrc_exec_t,s0)
 
 /usr/bin/zabbix_server	--	gen_context(system_u:object_r:zabbix_exec_t,s0)
+/usr/sbin/zabbix_agentd --	gen_context(system_u:object_r:zabbix_exec_t,s0)
 
 /var/log/zabbix(/.*)?		gen_context(system_u:object_r:zabbix_log_t,s0)
 
