diff -ur libselinux-2.1.9.orig/src/audit2why.c libselinux-2.1.9/src/audit2why.c
--- libselinux-2.1.9.orig/src/audit2why.c	2012-05-22 20:57:49.868101910 +0200
+++ libselinux-2.1.9/src/audit2why.c	2012-05-22 21:05:06.465065592 +0200
@@ -363,7 +363,7 @@
 		
 		/* make it a string */
 #if PY_MAJOR_VERSION >= 3
-		permstr = _PyUnicode_AsString( strObj );
+		permstr = PyBytes_AS_STRING(PyUnicode_AsEncodedString( strObj, "utf-8", "Error ~" ));
 #else
 		permstr = PyString_AsString( strObj );
 #endif
