--- sudo-1.6.7p5/auth/rfc1938.c	2003-04-16 01:39:15.000000000 +0100
+++ sudo-1.6.7p5/auth/rfc1938.c.new	2004-09-17 20:01:00.996902672 +0100
@@ -64,11 +64,7 @@
 #if defined(HAVE_SKEY)
 # include <skey.h>
 # define RFC1938				skey
-#  ifdef __NetBSD__
-#   define rfc1938challenge(a,b,c,d)	skeychallenge((a),(b),(c),(d))
-#  else
-#   define rfc1938challenge(a,b,c,d)	skeychallenge((a),(b),(c))
-#  endif
+#  define rfc1938challenge(a,b,c,d)	skeychallenge((a),(b),(c),(d))
 # define rfc1938verify(a,b)		skeyverify((a),(b))
 #elif defined(HAVE_OPIE)
 # include <opie.h>
