--- services/courier.fc	2010-08-03 15:11:05.000000000 +0200
+++ services/courier.fc	2011-03-03 16:07:02.664999997 +0100
@@ -6,7 +6,7 @@
 /usr/sbin/courierldapaliasd		--	gen_context(system_u:object_r:courier_exec_t,s0)
 /usr/sbin/couriertcpd			--	gen_context(system_u:object_r:courier_tcpd_exec_t,s0)
 
-/usr/lib(64)?/courier/authlib/.*	--	gen_context(system_u:object_r:courier_authdaemon_exec_t,s0)
+/usr/lib(64)?/courier/(courier-)?authlib/.*	--	gen_context(system_u:object_r:courier_authdaemon_exec_t,s0)
 /usr/lib(64)?/courier/courier/.*	--	gen_context(system_u:object_r:courier_exec_t,s0)
 /usr/lib(64)?/courier/courier/courierpop.* --	gen_context(system_u:object_r:courier_pop_exec_t,s0)
 /usr/lib(64)?/courier/courier/imaplogin --	gen_context(system_u:object_r:courier_pop_exec_t,s0)
@@ -16,7 +16,7 @@
 /usr/lib(64)?/courier/rootcerts(/.*)?		gen_context(system_u:object_r:courier_etc_t,s0)
 /usr/lib(64)?/courier/sqwebmail/cleancache\.pl -- gen_context(system_u:object_r:sqwebmail_cron_exec_t,s0)
 
-/var/lib/courier(/.*)?			--	gen_context(system_u:object_r:courier_var_lib_t,s0)
+/var/lib/courier(/.*)?				gen_context(system_u:object_r:courier_var_lib_t,s0)
 
 /var/run/courier(/.*)?			--	gen_context(system_u:object_r:courier_var_run_t,s0)
 
