--- apps/xfce4.te	1970-01-01 01:00:00.000000000 +0100
+++ apps/xfce4.te	2011-01-22 13:18:24.079000051 +0100
@@ -0,0 +1,2 @@
+policy_module(xfce, 0.0.1)
+
--- apps/xfce4.fc	1970-01-01 01:00:00.000000000 +0100
+++ apps/xfce4.fc	2011-01-22 13:22:10.602000045 +0100
@@ -0,0 +1,9 @@
+/usr/lib(64)?/xfce4/session/xfsm-shutdown-helper 	-- gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/session/balou-export-theme		-- gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/session/balou-install-theme 	-- gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/xfwm4/helper-dialog 	-- gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/xfconf/xfconfd 	-- gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/panel/wrapper 	-- gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/panel/migrate 	-- gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/exo-1/exo-helper-1 	-- gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/exo-1/exo-compose-mail-1 	-- gen_context(system_u:object_r:bin_t,s0)
