diff -ur libsemanage-2.1.6.orig/src/fcontext_record.c libsemanage-2.1.6/src/fcontext_record.c
--- libsemanage-2.1.6.orig/src/fcontext_record.c	2012-05-22 21:50:23.416071391 +0200
+++ libsemanage-2.1.6/src/fcontext_record.c	2012-05-22 22:32:47.786076823 +0200
@@ -48,6 +48,8 @@
 	tmp_key->expr = expr;
 	tmp_key->type = type;
 
+	fprintf(stderr, "key is %d (or %u)\n", tmp_key, tmp_key);
+
 	*key_ptr = tmp_key;
 	return STATUS_SUCCESS;
 }
@@ -84,6 +86,7 @@
 {
 
 	int rv = strcmp(fcontext->expr, key->expr);
+	fprintf(stderr, "Comparing \"%s\" with \"%s\" (key is %u)\n", fcontext->expr, key->expr, key);
 	if (rv != 0)
 		return rv;
 	else {
