diff -uNr setools-3.3.7.orig/Makefile.am setools-3.3.7/Makefile.am
--- setools-3.3.7.orig/Makefile.am	2012-05-19 20:17:51.853184751 +0200
+++ setools-3.3.7/Makefile.am	2012-05-19 20:22:35.991185357 +0200
@@ -10,7 +10,7 @@
 endif
 # sediffx is also built conditionally, from sediffx/Makefile.am
 
-SUBDIRS = libqpol libapol libsefs libpoldiff libseaudit secmds sechecker sediff man packages debian $(MAYBE_APOL) $(MAYBE_GUI) python
+SUBDIRS = libqpol libapol libsefs libpoldiff libseaudit secmds sechecker sediff man packages debian $(MAYBE_APOL) $(MAYBE_GUI)
 
 #old indent opts
 #INDENT_OPTS = -npro -nbad -bap -sob -ss -l132 -di1 -nbc -br -nbbb -c40 -cd40 -ncdb -ce -cli0 -cp40 -ncs -d0 -nfc1 -nfca -i8 -ts8 -ci8 -lp -ip0 -npcs -npsl -sc
diff -uNr setools-3.3.7.orig/configure.ac setools-3.3.7/configure.ac
--- setools-3.3.7.orig/configure.ac	2012-05-19 20:17:51.872184751 +0200
+++ setools-3.3.7/configure.ac	2012-05-19 20:19:40.789184983 +0200
@@ -217,8 +217,6 @@
    do_swigify=yes
 fi
 
-AM_PATH_PYTHON(2.7)
-
 AC_ARG_ENABLE(swig-python,
               AC_HELP_STRING([--enable-swig-python],
                              [build SWIG interfaces for Python]),
@@ -228,6 +226,7 @@
       AC_PROG_SWIG(2.0.0)
    fi
    SWIG_PYTHON
+   AM_PATH_PYTHON(2.7)
    do_swigify_python=yes
    do_swigify=yes
 fi
diff -uNr setools-3.3.7.orig/libqpol/src/module_compiler.h setools-3.3.7/libqpol/src/module_compiler.h
--- setools-3.3.7.orig/libqpol/src/module_compiler.h	2012-05-19 20:17:51.844184751 +0200
+++ setools-3.3.7/libqpol/src/module_compiler.h	2012-05-19 20:31:29.636186496 +0200
@@ -108,6 +108,7 @@
  * return -1. */
 int end_avrule_block(int pass);
 
+void append_filename_trans(filename_trans_rule_t * filename_trans_rules);
 #ifdef	__cplusplus
 }
 #endif
diff -uNr setools-3.3.7.orig/libqpol/src/policy_define.h setools-3.3.7/libqpol/src/policy_define.h
--- setools-3.3.7.orig/libqpol/src/policy_define.h	2012-05-19 20:17:51.845184751 +0200
+++ setools-3.3.7/libqpol/src/policy_define.h	2012-05-19 20:32:50.461186669 +0200
@@ -26,6 +26,7 @@
 avrule_t *define_cond_compute_type(int which);
 avrule_t *define_cond_pol_list(avrule_t * avlist, avrule_t * stmt);
 avrule_t *define_cond_te_avtab(int which);
+avrule_t *define_cond_filename_trans(void);
 cond_expr_t *define_cond_expr(uint32_t expr_type, void *arg1, void *arg2);
 int define_attrib(void);
 int define_av_perms(int inherits);
@@ -37,6 +38,7 @@
 int define_conditional(cond_expr_t * expr, avrule_t * t_list, avrule_t * f_list);
 int define_constraint(constraint_expr_t * expr);
 int define_dominance(void);
+int define_filename_trans(void);
 int define_fs_context(unsigned int major, unsigned int minor);
 int define_fs_use(int behavior);
 int define_genfs_context(int has_type);
